From 6c912f5cd2ab2a8c11e3c032a074df76c7c31055 Mon Sep 17 00:00:00 2001
From: Jetski <jetski@google.com>
Date: Tue, 21 Jul 2026 12:22:04 +0000
Subject: [PATCH] Patch Cargo.toml

---
 Cargo.toml | 17 ++---------------
 1 file changed, 2 insertions(+), 15 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 229ecbb..08e07b9 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -84,12 +84,10 @@ default = [
     "service",
     "ring",
     "jetstream",
-    "nkeys",
     "crypto",
     "object-store",
     "kv",
-    "websockets",
-    "nuid",
+    "chrono",
 ]
 experimental = []
 fips = [
@@ -97,7 +95,6 @@ fips = [
     "tokio-rustls/fips",
 ]
 jetstream = [
-    "dep:time",
     "dep:serde_nanos",
     "dep:tryhard",
     "dep:base64",
@@ -122,7 +119,6 @@ server_2_11 = []
 server_2_12 = []
 server_2_14 = []
 service = [
-    "dep:time",
     "dep:serde_nanos",
 ]
 slow_tests = []
@@ -357,7 +353,7 @@ version = "1.0"
 version = "1"
 
 [dependencies.rand]
-version = "0.10.1"
+version = "0.9.4"
 
 [dependencies.regex]
 version = "1.9.1"
@@ -394,15 +390,6 @@ version = "0.1.16"
 [dependencies.thiserror]
 version = "2.0"
 
-[dependencies.time]
-version = "0.3.36"
-features = [
-    "parsing",
-    "formatting",
-    "serde",
-    "serde-well-known",
-]
-optional = true
 
 [dependencies.tokio]
 version = "1.36"
-- 
2.55.0.229.g6434b31f56-goog

