From ae4b21146bb1b75bb27ef88c3af65ffeca33514d Mon Sep 17 00:00:00 2001
From: Jetski <jetski@google.com>
Date: Tue, 21 Jul 2026 12:35:34 +0000
Subject: [PATCH] Patch id_generator

---
 src/id_generator.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/id_generator.rs b/src/id_generator.rs
index e568416..25500cb 100644
--- a/src/id_generator.rs
+++ b/src/id_generator.rs
@@ -37,7 +37,7 @@ pub(crate) fn next() -> String {
 #[inline]
 pub(crate) fn next() -> String {
     use rand::distr::Alphanumeric;
-    use rand::{rng, RngExt};
+    use rand::{rng, Rng};
 
     rng()
         .sample_iter(Alphanumeric)
-- 
2.55.0.229.g6434b31f56-goog

