https://github.com/google/rust-p9/pull/13

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -8,6 +8,11 @@
 
 #![recursion_limit = "256"]
 
+// Non-cargo build systems may not automatically include this for tests, so
+// explicitly include it. We could also `#[cfg(not(test))]` `p9_wire_format`, but
+// that leads to warnings about unused imports.
+extern crate proc_macro;
+
 use proc_macro2::Span;
 use proc_macro2::TokenStream;
 use quote::quote;
