diff --git a/android/build.rs b/android/build.rs
index 4495ad56ca..9948b258d4 100644
--- a/android/build.rs
+++ b/android/build.rs
@@ -7,9 +7,8 @@ struct SqliteTypeChooser;
 
 impl ParseCallbacks for SqliteTypeChooser {
     fn int_macro(&self, name: &str, value: i64) -> Option<IntKind> {
-        if name.starts_with("SQLITE_TRACE_") {
-            Some(IntKind::U32)
-        } else if value >= i32::MIN as i64 && value <= i32::MAX as i64 {
+        if name.starts_with("SQLITE_TRACE_") { return Some(IntKind::U32); }
+        if value >= i32::MIN as i64 && value <= i32::MAX as i64 {
             Some(IntKind::I32)
         } else {
             None
