diff --git b/rules.mk a/rules.mk
index 078c0bb..3ccd69f 100644
--- b/rules.mk
+++ a/rules.mk
@@ -14,12 +14,14 @@
 	--cfg 'feature="alloc"'
 
 MODULE_LIBRARY_DEPS := \
+	trusty/user/base/lib/liballoc-rust \
 	$(call FIND_CRATE,bitflags) \
 	$(call FIND_CRATE,enumn) \
 	$(call FIND_CRATE,log) \
 	$(call FIND_CRATE,thiserror) \
 	$(call FIND_CRATE,zerocopy) \
 	trusty/user/base/lib/libcompiler_builtins-rust \
-	trusty/user/base/lib/libcore-rust
+	trusty/user/base/lib/libcore-rust \
+	external/trusty/lk/lib/rust_support \
 
 include make/library.mk
