diff --git a/src/parse.rs b/src/parse.rs
index e9bdb9da9..47bb89770 100644
--- a/src/parse.rs
+++ b/src/parse.rs
@@ -86,7 +86,7 @@ fn parse_protocol<R: BufRead>(mut reader: Reader<R>) -> Protocol {
                                 Ok(Event::GeneralRef(byte_ref)) => {
                                     if let Ok(Some(c)) = byte_ref.resolve_char_ref() {
                                         copyright.push(c);
-                                    } else if let Ok(content) = byte_ref.xml_content() {
+                                    } else if let Ok(content) = byte_ref.xml10_content() {
                                         if let Some(s) =
                                             quick_xml::escape::resolve_xml_entity(&content)
                                         {
